test/dynamo/test_activation_checkpointing.py

```diff
@@ -1994,6 +1999,188 @@ def forward(self, primals_1: "f32[4, 4]"):
         )
 
 
+class RematerializeACNodesPassTests(torch._dynamo.test_case.TestCase):
+    """Tests for AC reordering optimization in full graph (forward+backward in one graph)."""
+
+    def count_op(self, gm, target):
+        return sum(1 for n in gm.graph.nodes if n.target == target)
+
+    def _compile_and_capture(self, fn, remat_using_tags_for_fwd_loss_bwd_graph, inputs):
+        captured_gm = None
+
+        def compiler(gm, example_inputs):
+            nonlocal captured_gm
+            captured_gm = gm
+            return gm.forward
+
+        backend = aot_autograd(
+            fw_compiler=compiler,
```

Comment:
nit: you can pass in `inference_compiler=` instead of `fw_compiler` since you know you only want to test the inference codepath